src/main/scala/com/kinja/amqp/AmqpConsumer.scala

```diff
@@ -44,7 +51,15 @@ class AmqpConsumer(
 	 * @inheritdoc
 	 */
 	override def subscribe[A: Reads](timeout: FiniteDuration, spacing: FiniteDuration, processor: A => Future[Unit]): Unit = {
-		val listener = createListener(timeout, spacing, processor)
+		val rejecter = actorSystem.actorOf(Props(new Rejecter))
+
+		val proxy = actorSystem.actorOf(Props(new Proxy))
+		proxy ! SetListener(rejecter)
```

Comment:
I think this line is not necessary because `supervisor ! Connect` will set the listener to a real one anyways. If you're worried about what happens with RabbitMQ messages if the RabbitMQ connection is opened and reception starts before the `Connect` message is processed, this is not much better either, and `proxy` is in the `receive` state anyways so it will just discard any `Delivery` message before receiving a `SetListener`.

Reply:
There is a difference between a rejected message — which would probably be requeued immediately — and an ignored one, which would be requeued after timeout, and probably go through MySQL. With that approach we might see a brief increase of MySQL writes/reads, but it will stop after `rejecter` starts doing its job. Without it it would go on and on until the service is killed. Not a big deal since it's just 30 seconds, but still.

Reply:
I didn't mean to remove `Rejecter` altogether. I only meant that setting the listener in `proxy` to `rejecter` (line 57) just a few lines before sending the `Connect` message to `supervisor` (line 62), which very soon sets `proxy` to a real listener, might not make much difference at the time of initializing the RabbitMQ subscription.

Reply:
Make it even clearer: at the time of running the code of the `subscribe` function, this
```
proxy ! SetListener(rejecter)
...
supervisor ! Connect
...
proxy ! SetListener(listener)
```
is essentially the same as
```
// keeping proxy in the default receive loop
...
supervisor ! Connect
...
proxy ! SetListener(listener)

```

Reply:
Oh, I see. Thanks.

Reply:
BTW, this can even hurt, since messages from different actors are not guaranteed to arrive in order. Imagine `SetListener(rejecter)`, by some weird chance, arriving later than `SetListener(listener)`.